src/App.js

```diff
@@ -1,56 +1,125 @@
-import React, { useState, useEffect } from "react";
+import React, {useEffect, useState} from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 function App() {
-  const [todoList, setTodoList] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
-
-  useEffect(() => {
-    new Promise ((resolve, reject) => 
-      setTimeout(() => 
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
-          },
-        }),
-       2000
-       )
-    ).then((result) => {
-      setTodoList(result.data.todoList);
-      setIsLoading(false);
-    });
-  }, []);
-
-  useEffect(() => {
-    if (!isLoading) {
-      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-    }
-
-  }, [todoList, isLoading]);
-
-  function addTodo(newTodo) {
-    setTodoList((prevTodos) => [...prevTodos, newTodo]);
-  }
-
-  const removeTodo = (id) => {
-    const updateList = todoList.filter((todo) => todo.id !== id);
-    setTodoList(updateList);
-  };
-
-  return (
-    <>
-      <h1>ToDo List</h1>
-      <AddTodoForm onAddTodo={addTodo} />
-      {
-        isLoading ? (
-          <p>Loading ...</p>
-        ):(
-        <TodoList todoList={todoList} onRemoveTodo={removeTodo}/>
-        )
-      }
-    </>
-  );
+    const [todoList, setTodoList] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+
+    // Fetch data from Airtable API
+    const fetchData = async () => {
+        const options = {
+            method: 'GET',
+            headers: {
+                'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`
+            },
+        };
+
+        const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+        try {
+            const response = await fetch(url, options);
+            if (!response.ok) {
+                throw new Error(`Error: ${response.status}`);
+            }
+
+            const data = await response.json();
+
+            // Mapping data
+            const todos = data.records.map(record => ({
+                title: record.fields.title,
+                id: record.id,
+                completed: record.fields.completed || false,
+            }));
+
+            setTodoList(todos); // Set todoList
+            setIsLoading(false); // indicate that the fetch is complete
+
+        } catch (error) {
+            console.error('Fetch error:', error.message);
+        } finally {
+            setIsLoading(false);
+        }
```

Comment:
Here you have `setIsLoading(false)` twice, which is unnecessary. Since you're using a `finally` block you only need the state update there. 
The flow is: 

1. `try` to get the data, if successful update the `todoList` state, `finally` update the loading state. 
2. `try` to get the data, if unsuccessful `catch` the error and display it, `finally` update the loading state. 
```suggestion
        try {
            const response = await fetch(url, options);
            if (!response.ok) {
                throw new Error(`Error: ${response.status}`);
            }

            const data = await response.json();

            // Mapping data
            const todos = data.records.map(record => ({
                title: record.fields.title,
                id: record.id,
                completed: record.fields.completed || false,
            }));

            setTodoList(todos); // Set todoList

        } catch (error) {
            console.error('Fetch error:', error.message);
        } finally {
            setIsLoading(false);
        }
```